src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponentResult.cs

```diff
@@ -0,0 +1,105 @@
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ViewComponentResult : ActionResult
+    {
+        /// <summary>
+        /// Gets or sets the arguments provided to the view component.
+        /// </summary>
+        public object[] Arguments { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="MediaTypeHeaderValue"/> representing the Content-Type header of the response.
+        /// </summary>
+        public MediaTypeHeaderValue ContentType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the HTTP status code.
+        /// </summary>
+        public int? StatusCode { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ITempDataDictionary"/> for this result.
+        /// </summary>
+        public ITempDataDictionary TempData { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the view component to invoke. Will be ignored if <see cref="ViewComponentType"/>
+        /// is set to a non-<c>null</c> value.
+        /// </summary>
+        public string ViewComponentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the type of the view component to invoke.
+        /// </summary>
+        public Type ViewComponentType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ViewDataDictionary"/> for this result.
+        /// </summary>
+        public ViewDataDictionary ViewData { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IViewEngine"/> used to locate views.
+        /// </summary>
+        /// <remarks>When <c>null</c>, an instance of <see cref="ICompositeViewEngine"/> from
+        /// <c>ActionContext.HttpContext.RequestServices</c> is used.</remarks>
+        public IViewEngine ViewEngine { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
+        {
+            var response = context.HttpContext.Response;
+            var services = context.HttpContext.RequestServices;
+
+            var htmlHelperOptions = services.GetRequiredService<IOptions<MvcViewOptions>>().Options.HtmlHelperOptions;
+            var viewComponentHelper = services.GetRequiredService<IViewComponentHelper>();
+
+            var contentTypeHeader = ContentType ?? ViewExecutor.DefaultContentType;
+            if (contentTypeHeader.Encoding == null)
+            {
+                // Do not modify the user supplied content type, so copy it instead
+                contentTypeHeader = contentTypeHeader.Copy();
+                contentTypeHeader.Encoding = Encoding.UTF8;
+            }
+
+            using (var writer = new HttpResponseStreamWriter(response.Body, contentTypeHeader.Encoding))
+            {
+                var viewContext = new ViewContext(
+                    context,
+                    NullView.Instance,
```

Comment:
Thoughts on null-object pattern here? I don't like the idea of allowing the `view` argument to be null.